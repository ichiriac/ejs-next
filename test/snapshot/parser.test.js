/**
 * Copyright (C) 2019 Ioan CHIRIAC (MIT)
 * @authors https://github.com/ichiriac/ejs2/graphs/contributors
 * @url https://ejs.js.org
 */
"use strict";

const parser = require('../../lib/parser/index');

describe('Parser', () => {
  it('Simple case', () => {
    var reader = new parser();
    var program = reader.eval(`
    <foo> <%%
    <%# comment %>
    <%_ if (foo == "%>\\"%>") { _%>
      <%= bar %>
    <% } %>
    </foo>
    `);
    expect(program).toMatchSnapshot();
  });
});