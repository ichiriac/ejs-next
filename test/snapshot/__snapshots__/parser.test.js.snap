// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser Simple case 1`] = `
Object {
  "childs": Array [
    Object {
      "kind": "inline",
      "location": Object {
        "end": Object {
          "col": 5,
          "line": 3,
          "offset": 19,
        },
        "src": "
    <foo> <%%
    ",
        "start": Object {
          "col": 0,
          "line": 1,
          "offset": 0,
        },
      },
      "source": "
    <foo> <%%
    ",
    },
    Object {
      "kind": "comment",
      "location": Object {
        "end": Object {
          "col": 20,
          "line": 3,
          "offset": 33,
        },
        "src": "<%# comment %>",
        "start": Object {
          "col": 5,
          "line": 3,
          "offset": 19,
        },
      },
      "source": " comment ",
    },
    Object {
      "kind": "inline",
      "location": Object {
        "end": Object {
          "col": 5,
          "line": 4,
          "offset": 38,
        },
        "src": "
    ",
        "start": Object {
          "col": 20,
          "line": 3,
          "offset": 33,
        },
      },
      "source": "
    ",
    },
    Object {
      "kind": "statement",
      "location": Object {
        "end": Object {
          "col": 36,
          "line": 4,
          "offset": 68,
        },
        "src": "<%_ if (foo == \\"%>\\\\\\"%>\\") { _%>",
        "start": Object {
          "col": 5,
          "line": 4,
          "offset": 38,
        },
      },
      "postStrip": true,
      "preStrip": true,
      "source": " if (foo == \\"%>\\\\\\"%>\\") { ",
    },
    Object {
      "kind": "inline",
      "location": Object {
        "end": Object {
          "col": 7,
          "line": 5,
          "offset": 75,
        },
        "src": "
      ",
        "start": Object {
          "col": 36,
          "line": 4,
          "offset": 68,
        },
      },
      "source": "
      ",
    },
    Object {
      "kind": "output",
      "location": Object {
        "end": Object {
          "col": 18,
          "line": 5,
          "offset": 85,
        },
        "src": "<%= bar %>",
        "start": Object {
          "col": 7,
          "line": 5,
          "offset": 75,
        },
      },
      "postStrip": false,
      "sanitize": true,
      "source": " bar ",
    },
    Object {
      "kind": "inline",
      "location": Object {
        "end": Object {
          "col": 5,
          "line": 6,
          "offset": 90,
        },
        "src": "
    ",
        "start": Object {
          "col": 18,
          "line": 5,
          "offset": 85,
        },
      },
      "source": "
    ",
    },
    Object {
      "kind": "statement",
      "location": Object {
        "end": Object {
          "col": 13,
          "line": 6,
          "offset": 97,
        },
        "src": "<% } %>",
        "start": Object {
          "col": 5,
          "line": 6,
          "offset": 90,
        },
      },
      "postStrip": false,
      "preStrip": false,
      "source": " } ",
    },
    Object {
      "kind": "inline",
      "location": Object {
        "end": Object {
          "col": 4,
          "line": 8,
          "offset": 113,
        },
        "src": "
    </foo>
    ",
        "start": Object {
          "col": 13,
          "line": 6,
          "offset": 97,
        },
      },
      "source": "
    </foo>
    ",
    },
  ],
  "kind": "program",
  "location": Object {
    "end": Object {
      "col": 4,
      "line": 8,
      "offset": 113,
    },
    "src": "
    <foo> <%%
    <%# comment %>
    <%_ if (foo == \\"%>\\\\\\"%>\\") { _%>
      <%= bar %>
    <% } %>
    </foo>
    ",
    "start": Object {
      "col": 0,
      "line": 1,
      "offset": 0,
    },
  },
}
`;
